### PR TITLE
Update message recovery command

### LINKED ISSUE #7727: Update message recovery command
### Description

Update message recovery command based on [LIP0054](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#state-recovery-command)
- Update [schema](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#parameters-1) for params
- Update [verification](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#verification-1)
- Update [execution](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#execution-1)

### Acceptance Criteria

- Should have all the unit tests

### Additional Information

- Related LIP PR https://github.com/LiskHQ/lips/pull/172
- Related LIP https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md

### PR DESCRIPTION
**Pull request by sitetester**
### What was the problem?

This PR resolves #7727

### How was it solved?

Common code moved to BaseMessageRecoveryCommand

### How was it tested?
- test cases moved to single file & further duplication issues addressed
- still work In progress...

### CODE REVIEW

File: framework/src/modules/interoperability/mainchain/commands/message_recovery.ts [hunk 1]
```diff
@@ -41,129 +39,93 @@ export class MainchainMessageRecoveryCommand extends BaseInteroperabilityCommand
...
+			const isLive = await this.internalMethod.isLive(context, ccm.sendingChainID, Date.now());
+			if (!isLive) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Cross-chain message sending chain is not live.'),
+				};
 			}
+		}
+
+		return verifyMessageRecovery(context.params, terminatedChainOutboxAccount);
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/message_recovery.ts:77]: I don't feel that having this function is much helping us as we only have this code in this verify function and also we are looping twice over the deserialized CCMs just for isLive function. Maybe move all the logic to this place as we are not using it in sidehchain module.

File: framework/src/modules/interoperability/mainchain/commands/message_recovery.ts [hunk 2]
```diff
@@ -41,129 +39,93 @@ export class MainchainMessageRecoveryCommand extends BaseInteroperabilityCommand
...
 
-			const recoveryCCM: CCMsg = {
-				...ccm,
-				fee: BigInt(0),
-				status: CCMStatusCode.RECOVERED,
+	// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#execution-1
+	public async execute(context: CommandExecuteContext<MessageRecoveryParams>): Promise<void> {
+		const { params } = context;
+		// Set CCM status to recovered and assign fee to trs sender.
+		// const crossChainMessages = params.crossChainMessages.map(ccmBytes => codec.decode<CCM>(ccmSchema, ccmBytes))
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/message_recovery.ts]: we can remove this

File: framework/src/modules/interoperability/mainchain/commands/message_recovery.ts [hunk 3]
```diff
@@ -41,129 +39,93 @@ export class MainchainMessageRecoveryCommand extends BaseInteroperabilityCommand
...
 		}
 
-		const doesTerminatedOutboxAccountExist = await this.stores
+		const terminatedOutboxAccount = await this.stores
 			.get(TerminatedOutboxStore)
-			.has(context, chainIdAsBuffer);
-
-		if (!doesTerminatedOutboxAccountExist) {
+			.get(context, context.params.chainID);
+		if (!terminatedOutboxAccount) {
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/message_recovery.ts:110]: we don't need this check as its already checked in verify

File: framework/src/modules/interoperability/mainchain/commands/message_recovery.ts [hunk 4]
```diff
@@ -41,129 +39,93 @@ export class MainchainMessageRecoveryCommand extends BaseInteroperabilityCommand
...
-
-			const chainAccount = await this.stores.get(ChainAccountStore).get(context, ccmChainId);
-
-			if (chainAccount.status !== ChainStatus.ACTIVE) {
-				continue;
-			}
-
-			await this.internalMethod.addToOutbox(context, ccmChainId, newCcm);
-		}
+		terminatedOutboxAccount.outboxRoot = regularMerkleTree.calculateRootFromUpdateData(
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/message_recovery.ts:121]: you also need to set this value back in the TerminatedOutboxStore store

**Review by ishantiw — CHANGES_REQUESTED** [review]
We need to remove message recovery from sidechain and associated code.

File: framework/src/modules/interoperability/mainchain/commands/message_recovery.ts [hunk 5]
```diff
@@ -20,150 +20,185 @@ import {
...
-			terminatedChainOutboxAccount = await this.stores
-				.get(TerminatedOutboxStore)
-				.get(context, chainIdAsBuffer);
+			terminatedOutboxAccount = await this.stores.get(TerminatedOutboxStore).get(context, chainID);
 		} catch (error) {
 			if (!(error instanceof NotFoundError)) {
 				throw error;
 			}
 		}
+		if (!terminatedOutboxAccount) {
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/message_recovery.ts:62]: you can move this condition inside catch above, 

```ts
catch (error) {
		if (error instanceof NotFoundError)) {
			return {
				status: VerifyStatus.FAIL,
				error: new Error('Terminated outbox account does not exist.'),
			};
		}
		throw error
	}
```

File: framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts [hunk 6]
```diff
@@ -436,119 +590,40 @@ describe('Mainchain MessageRecoveryCommand', () => {
...
-			for (const ccm of ccms) {
-				const chainID = ccm.sendingChainID;
-
-				when(command['internalMethod'].isLive as never)
-					.calledWith(chainID)
-					.mockResolvedValue(false as never);
-			}
-
-			await command.execute(commandExecuteContext);
+		it('should proceed without error if the sending chain is the mainchain', async () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts:593]: should call applyRecovery ? and also check applyRecovery to have been called with right params?

File: framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts [hunk 7]
```diff
@@ -436,119 +590,40 @@ describe('Mainchain MessageRecoveryCommand', () => {
...
-
-				when(chainAccountStoreMock.get)
-					.calledWith(chainID)
-					.mockResolvedValue({
-						status: -1,
-					} as any);
-			}
-
-			await command.execute(commandExecuteContext);
+		it('should proceed without error if the sending chain is not the mainchain', async () => {
```
Comment (ishantiw) [inline framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts:611]: same as above, 
should call forwardRecovery ? and also check forwardRecovery to have been called with right params?

Comment (ishantiw) [reply framework/test/unit/modules/interoperability/mainchain/commands/message_recovery.spec.ts:611]: also check that terminatedOutboxAccount.set is called or it has been set if in memory calling `.has`? same below

**Review by ishantiw — APPROVED** [review]
Overall looks fine to me, few more comments

**Review by ishantiw — CHANGES_REQUESTED** [review]
Updating to review changes

**Review by mitsuaki-u — APPROVED** [review]
Well documented comments :)

File: framework/src/modules/interoperability/mainchain/commands/message_recovery.ts [hunk 8]
```diff
@@ -20,150 +20,183 @@ import {
...
+				};
+			}
+		}
 
-		const updatedCCMs: Buffer[] = [];
-		const deserializedCCMs = params.crossChainMessages.map(serializedCCMsg =>
-			codec.decode<CCMsg>(ccmSchema, serializedCCMsg),
+		// Process basic checks for all CCMs.
+		// Verify general format. Past this point, we can access ccm root properties.
+		const deserializedCCMs = crossChainMessages.map(serializedCcm =>
```
Comment (ishantiw) [inline framework/src/modules/interoperability/mainchain/commands/message_recovery.ts:93]: we can move decoding part within the loop below

**Review by ishantiw — CHANGES_REQUESTED** [review]
Just one more comment and it should be good

**Review by ishantiw — APPROVED** [review]
🎉

**Pull request merged**